packages/trie/examples/README.md

```diff
@@ -0,0 +1,819 @@
+```
+
+Indeed! A leaf node with value "testValue0". The "nibble" indicates the last hex character(s) that differentiate this leaf from the parent branch. Since our branch node was "testKey" (in bytes = `<74 65 73 74 56 61 6c 75 65>`) , and since we took the branch corresponding to hex value `3`, we only need the last `0` to complete our full key/path of "testKey0" (in bytes = `<74 65 73 74 56 61 6c 75 65 30>`).
+
+### Example 2d - Creating and Looking Up an Extension Node
+
+To create an extension node, we need to slightly change our keys. We'll keep our branch node at path "testKey", but we'll change the two other keys so that they lead down a lengthy common path.
+
+```jsx
+console.log(Buffer.from('testKey'))
+console.log(Buffer.from('testKey0001'))
+console.log(Buffer.from('testKey000A'))
+
+// RESULT
+<Buffer 74 65 73 74 4b 65 79>
+<Buffer 74 65 73 74 4b 65 79 30 30 30 31>
+<Buffer 74 65 73 74 4b 65 79 30 30 30 41>
+```
+
+As you can see, the bytes `<30 30 30>` (standing for `000`> are common to both keys. We therefore should assume an extension that begins at index `3` of the branch node at "testKey". Let's see:
+
+```jsx
+await trie.put(Buffer.from('testKey'), Buffer.from('testValue'))
+await trie.put(Buffer.from('testKey0001'), Buffer.from('testValue0'))
+await trie.put(Buffer.from('testKey000A'), Buffer.from('testValueA'))
+
+var node1 = await trie.findPath(Buffer.from('testKey')) // Looking up our branch node
+console.log(node1.node) // The branch node
+
+// RESULT
+
+BranchNode {
+  _branches: [
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer 39 1d 48 30 de 00 87 57 46 98 4c 4f d3 ef 5a 0c f7 ca 7b 40 f9 c2 8a ce e2 ba 22 49 84 41 8f 6b>,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >,
+    <Buffer >
+  ],
+  _value: <Buffer 74 65 73 74 56 61 6c 75 65>
+}
+```
+
+As we should expect, there is a branch at index `3`. The branch contains a hash that we can use to lookup the child node:
+
+```jsx
+ var node2 = await trie.lookupNode(Buffer.from(node1.node._branches[3])) // Looking up the child node from its hash (using lookupNode)
+ console.log(node2) // An extension node!
+
+// RESULT
+ExtensionNode {
+  _nibbles: [ 0, 3, 0, 3, 0 ],
+  _value: <Buffer 70 b3 d0 20 ad 85 8f d6 00 28 a4 23 e9 8f 1d 99 c5 37 cd b9 1f 27 49 16 40 06 6f ea c7 9c 2f 72>
+}
+```
+
+Awesome! An extension node!
+
+You might have noticed that this child node is a "hash", while in the previous examples, it was the leaf node itself. This is simply because of the way nodes are stored in branch nodes. If the [Recursive Length Prefix](https://github.com/ethereum/wiki/wiki/RLP) (RLP) encoding of the child node is less than 32 bytes, the node is stored directly. However, if the RLP encoding is equal to or longer than 32 bytes, a hash of the node is stored in the branch node, which can be used to lookup the child node directly.
```

Comment:
```suggestion
You might have noticed that this child node is a "hash", while in the previous examples, it was the leaf node itself. This is simply because of the way nodes are stored in branch nodes. If the [Recursive Length Prefix](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp) (RLP) encoding of the child node is less than 32 bytes, the node is stored directly. However, if the RLP encoding is equal to or longer than 32 bytes, a hash of the node is stored in the branch node, which can be used to lookup the child node directly.
```